sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -47,31 +47,28 @@ class MapboxMap :
   MapCameraManagerDelegate,
   MapStyleStateDelegate {
 
-  private var nativeMap: MapInterface?
+  private val nativeMap: MapInterface
```

Comment:
If users hold the reference to MapboxMap, it will also keep nativeMap and maybe have memory leak issue.

Reply:
that's totally correct - that's the whole purpose of this PR. If users hold the ref to `MapboxMap` in their codebase and try to access any method after `MapView.onDestroy()` is triggered - we would print an error log for that. And yes, that's a leak, but not a leak from our end, it's a clear leak from user's end, we just clearly state that but not throw any exceptions as this could be a breaking behaviour change. Accessing native object after map view and renderer are destroyed should not bring in a crash, if it does in some cases - that should be fixed in gl-native side

Reply:
Let's add documentation for that.
Add a note at to the top-level of the file, also to the `MapView.getMapboxMap` highlighting that MapboxMap keeps a lot of memory with it and should NOT be cached outside of the activity.